spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessorTests.java

```diff
@@ -195,6 +203,33 @@ public void invalidValueInAnnotationTestBean() {
 		}
 	}
 
+	@Test
+	public void multipleBindingsTestBean() {
+		ConfigurableApplicationContext context = new AnnotationConfigApplicationContext(Config.class,
+				MultipleBindingsTestBean.class);
+
+		RabbitListenerContainerTestFactory factory = context.getBean(RabbitListenerContainerTestFactory.class);
+		assertThat("one container should have been registered", factory.getListenerContainers(), hasSize(1));
+		RabbitListenerEndpoint endpoint = factory.getListenerContainers().get(0).getEndpoint();
+		assertEquals(Collections.singletonList("my_queue"),
+				((AbstractRabbitListenerEndpoint) endpoint).getQueueNames());
+		final List<Queue> queues = new ArrayList<>(context.getBeansOfType(Queue.class).values());
+		queues.sort(Comparator.comparing(Queue::getName));
+		assertThat(queues.stream().map(Queue::getName).collect(Collectors.toList()),
+				contains("my_queue", "secondQueue", "testQueue"));
```

Comment:
This check can be simplified by moving `testQueue`/`secondQueue` from `Config` into another class, which is included only when needed. Can be a separate PR. Should I?